api/pkg/apis/v1alpha1/model/campaign.go

```diff
@@ -142,9 +142,6 @@ func (c CampaignSpec) DeepEquals(other IDeepEquals) (bool, error) {
 		return false, nil
 	}
 
-	if len(c.Stages) != len(otherC.Stages) {
```

Comment:
I think it's desirable to keep this to fail faster if the stage numbers don't match up

Reply:
This is duplicated with the logic above.

https://github.com/eclipse-symphony/symphony/blob/38d9d8c1b0d09b595e39cb936d75ca60313fffbb/api/pkg/apis/v1alpha1/model/campaign.go#L141-L147